src/mapgen.cpp

```diff
@@ -6311,8 +6311,43 @@ std::vector<item *> map::place_items(
             tries++;
         } while( is_valid_terrain( p ) && tries < 20 );
         if( tries < 20 ) {
-            auto put = put_items_from_loc( group_id, p, turn );
-            res.insert( res.end(), put.begin(), put.end() );
+            for( const item &itm : item_group::items_from( group_id, turn, spawn_flags::use_spawn_rate ) ) {
+                const float item_cat_spawn_rate = std::max( 0.0f, item_category_spawn_rate( itm ) );
+                // for items with category spawn rate less than or equal to 1, roll a dice to see if they should spawn
+                if( item_cat_spawn_rate <= 1 ) {
+                    if( rng_float( 0.1f, 1 ) <= item_cat_spawn_rate ) {
```

Comment:
I think rng_float( 0.1, 1.0) does uniform distribution? By starting from 0.1 the actual spawn rate will deviate from the expected category setting (the closer you set it to 0.1) since this is a direct comparison here.

For example: if I set Guns spawn rate to 0.15 I would expect them to be 15% rarer that usually. But rng_float( 0.1, 1.0 ) will roll a value <= 0.15 only in approx 5/90 = 5.6% cases making the guns 3 times rarer than my setting says.

Is there a reason to start the roll from 0.1 and not from zero?

Reply:
I reserved setting to 0 to fully forbid spawning items from category in question.